platform/web/js/libs/library_godot_javascript_singleton.js

```diff
@@ -81,11 +81,21 @@ const GodotJSWrapper = {
 			case 0:
 				return null;
 			case 1:
-				return !!GodotRuntime.getHeapValue(val, 'i64');
-			case 2:
-				return GodotRuntime.getHeapValue(val, 'i64');
+				return Boolean(GodotRuntime.getHeapValue(val, 'i64'));
+			case 2: {
+				const heap_value = GodotRuntime.getHeapValue(val, 'i64');
+				switch (typeof heap_value) {
+				case 'bigint':
+					return heap_value > Number.MAX_SAFE_INTEGER || heap_value < Number.MIN_SAFE_INTEGER
+						? heap_value
+						: Number(heap_value);
+				case 'number':
+				default:
+					return heap_value;
+				}
+			}
```

Comment:
@Faless, I implemented your suggestions with added checks as Godot may be built without bigint support (old emscripten versions).

Reply:
I don't think we really need the switch/case.

If the returned value is a Number (old emcc) and not a safe integer, we'd simply needlessly cast it to Number (which shoud be noop).
This is trading two number comparisons, for a typeof call + string comparison (which should be slower, though with JIT it might end up having the same performance).

Then again, not a big deal, so I think this is also fine.